test/destructive_schema_test.dart

```diff
@@ -0,0 +1,86 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+@Tags(['destructive'])
+import 'dart:async';
+import 'package:test/test.dart' hide test, throws;
+import '../lib/realm.dart';
+import 'test.dart';
+
+part 'destructive_schema_test.g.dart';
+
+@RealmModel()
+@MapTo("Card")
+class _CardV1 {
+  @PrimaryKey()
+  @MapTo('_id')
+  late ObjectId id;
+}
+
+@RealmModel()
+@MapTo("Card")
+class _CardV2 {
+  @PrimaryKey()
+  @MapTo('_id')
+  late Uuid id;
+}
+
+// Destructive schema change causes the server to force the translator to restart.
```

Comment:
Can this fail negatively impact the other workflows running in parallel and reusing the apps?

Reply:
Yes, see above.

Reply:
How are we making sure we don't impact other workflows here?